client/components/report/AttributeFilterDialog.tsx

```
@@ -0,0 +1,330 @@
+      return initialAttributeTypes;
+    }
+    
+    // Otherwise calculate them
+    const typeMap: Record<string, 'numeric' | 'categorical'> = {};
+    Object.entries(availableAttributes).forEach(([attribute, values]) => {
+      const isNumeric = values.every(value => {
+        const trimmedValue = value.trim();
+        return trimmedValue === '' || !isNaN(Number(trimmedValue));
+      });
+      typeMap[attribute] = isNumeric && values.length > 0 ? 'numeric' : 'categorical';
+    });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)